src/shared_modules/content_manager/src/components/gzipDecompressor.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Wazuh content manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * October 20, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _GZIP_DECOMPRESSOR_HPP
+#define _GZIP_DECOMPRESSOR_HPP
+
+#include "json.hpp"
+#include "updaterContext.hpp"
+#include "utils/chainOfResponsability.hpp"
+#include "utils/stringHelper.h"
+#include "utils/zlibHelper.hpp"
+#include <filesystem>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+
+/**
+ * @class GzipDecompressor
+ *
+ * @brief Class in charge of decompressing the content in GZ format as a step of a chain of responsibility.
+ *
+ */
+class GzipDecompressor final : public AbstractHandler<std::shared_ptr<UpdaterContext>>
+{
+private:
+    /**
+     * @brief Pushes the state of the current stage into the data field of the context.
+     *
+     * @param contextData Reference to the context data.
+     * @param status Status to be pushed.
+     */
+    void pushStageStatus(nlohmann::json& contextData, std::string status) const
+    {
+        auto statusObject = nlohmann::json::object();
+        statusObject["stage"] = "GzipDecompressor";
+        statusObject["status"] = std::move(status);
+
+        contextData.at("stageStatus").push_back(std::move(statusObject));
+    }
+
+    /**
+     * @brief Decompress the compressed content and update the context paths.
+     *
+     * @param context Updater context.
+     */
+    void decompress(UpdaterContext& context) const
+    {
+        for (auto& path : context.data.at("paths"))
+        {
+            // Copy input path.
+            std::filesystem::path inputPath {path};
+            auto outputPath {path.get<std::string>()};
+
+            // Replace downloads folder for contents folder.
+            const auto& outputFolder {context.spUpdaterBaseContext->outputFolder};
+            Utils::replaceFirst(
+                outputPath, (outputFolder / DOWNLOAD_FOLDER).string(), (outputFolder / CONTENTS_FOLDER).string());
```

Comment:
Nitpick (complexity): Could you add and example as a comment? Just to better understand the motivation

Question: Where are DOWNLOAD_FOLDER and CONTENTS_FOLDER set?

Reply:
Okay okay, seems about right

Reply:
> Nitpick (complexity): Could you add and example as a comment? Just to better understand the motivation

Done.